src/objective-c/tests/UnitTests/TransportTests.m

```diff
@@ -0,0 +1,232 @@
+
+@implementation DummyInterceptorFactory
+
++ (instancetype)sharedInstance {
+  dispatch_once(&initDummyInterceptorFactory, ^{
+    dummyInterceptorFactory = [[DummyInterceptorFactory alloc] init];
+  });
+  return dummyInterceptorFactory;
+}
+
+- (GRPCInterceptor *)createInterceptorWithManager:(GRPCInterceptorManager *)interceptorManager {
+  return [[DummyInterceptor alloc]
+      initWithInterceptorManager:interceptorManager
+                   dispatchQueue:dispatch_queue_create(NULL, DISPATCH_QUEUE_SERIAL)];
+}
+
+@end
+
+@interface TestsBlockCallbacks : NSObject<GRPCResponseHandler>
+
+- (instancetype)initWithInitialMetadataCallback:(void (^)(NSDictionary *))initialMetadataCallback
+                                   dataCallback:(void (^)(id))dataCallback
+                                  closeCallback:(void (^)(NSDictionary *, NSError *))closeCallback
+                           writeMessageCallback:(void (^)(void))writeMessageCallback;
+
+@end
+
+@implementation TestsBlockCallbacks {
+  void (^_initialMetadataCallback)(NSDictionary *);
+  void (^_dataCallback)(id);
+  void (^_closeCallback)(NSDictionary *, NSError *);
+  void (^_writeMessageCallback)(void);
+  dispatch_queue_t _dispatchQueue;
+}
+
+- (instancetype)initWithInitialMetadataCallback:(void (^)(NSDictionary *))initialMetadataCallback
+                                   dataCallback:(void (^)(id))dataCallback
+                                  closeCallback:(void (^)(NSDictionary *, NSError *))closeCallback
+                           writeMessageCallback:(void (^)(void))writeMessageCallback {
+  if ((self = [super init])) {
+    _initialMetadataCallback = initialMetadataCallback;
+    _dataCallback = dataCallback;
+    _closeCallback = closeCallback;
+    _writeMessageCallback = writeMessageCallback;
+    _dispatchQueue = dispatch_queue_create(nil, DISPATCH_QUEUE_SERIAL);
+  }
+  return self;
+}
+
+- (void)didReceiveInitialMetadata:(NSDictionary *)initialMetadata {
+  if (_initialMetadataCallback) {
+    _initialMetadataCallback(initialMetadata);
+  }
+}
+
+- (void)didReceiveProtoMessage:(id)message {
+  if (_dataCallback) {
+    _dataCallback(message);
+  }
+}
+
+- (void)didCloseWithTrailingMetadata:(NSDictionary *)trailingMetadata error:(NSError *)error {
+  if (_closeCallback) {
+    _closeCallback(trailingMetadata, error);
+  }
+}
+
+- (void)didWriteMessage {
+  if (_writeMessageCallback) {
+    _writeMessageCallback();
+  }
+}
+
+- (dispatch_queue_t)dispatchQueue {
+  return _dispatchQueue;
+}
+@end
+
+@interface TransportTests : XCTestCase
+
+@end
+
+@implementation TransportTests
+
+- (void)testTransportInterceptors {
+  __weak XCTestExpectation *expectComplete =
+      [self expectationWithDescription:@"Expect call complete"];
+  [GRPCFakeTransportFactory sharedInstance].nextTransportInstance = nil;
+
+  [[GRPCFakeTransportFactory sharedInstance]
+      setTransportInterceptorFactories:@[ [DummyInterceptorFactory sharedInstance] ]];
```

Comment:
multiple transport interceptors are supported, so a test with 2 interceptors might be useful (first interceptor is passthrough, second intereptor cancels the call).